src/gui/qgsfilewidget.cpp

```diff
@@ -220,6 +228,43 @@ QgsFilterLineEdit *QgsFileWidget::lineEdit()
   return mLineEdit;
 }
 
+void QgsFileWidget::updateLayout()
+{
+  mLayout->removeWidget( mLineEdit );
```

Comment:
I'm not 100% sure (and the documentation is unclear), but I *think* this results in a leak of the QLayoutItem associated with the widget.

Reply:
I think this is OK: 

> Note: The ownership of widget remains the same as when it was added.

https://doc.qt.io/qt-5/qlayout.html#removeWidget